src/allotropy/parsers/utils/pandas.py

```diff
@@ -88,6 +90,31 @@ def assert_value_from_df(df: pd.DataFrame, key: str) -> Any:
         raise AllotropeConversionError(msg) from e
 
 
+def parse_header_row(df: pd.DataFrame) -> pd.DataFrame:
+    # Set the first row of a dataframe as the columns. This is useful when the dataframe has already been
+    # parsed (e.g. when splitting a dataframe into two parts), and so the header/index_col arguments cannot
+    # be used to set the columns/index at read time.
+    df.columns = pd.Index(
+        assert_not_empty_df(df, "Cannot set parse header row for empty dataframe.")
+        .astype(str)
+        .iloc[0]
+    )
+    df.columns = df.astype(str).columns.str.strip()
+    return df[1:]
+
+
+def split_header_and_data(
+    df: pd.DataFrame, cond: Callable[[pd.Series[Any]], bool]
```

Comment:
nit: could you rename `cond` to be more descriptive? something like `should_split_on_row`